### DISCUSSION TITLE
Nuxt 3 - SSR - Cache Cloudflare

### DISCUSSION THREAD

**Discussion by nhatimme** [nuxt/nuxt#20804] [General]
I have a specific question about NUXT 3 and the possibilities for our solution.

NUXT 3 (SSR Enabled)

We have created an E-commerce webshop. The pages, products, categories are generated dynamically. That is, when someone navigates to: url.com/chairs (directly) the server validates whether this slug exists (the possible slugs come from an external api endpoint). If it exists it will then fetch and display the data for the page. This is going well.

However, we are now trying to implement the following, which we are currently struggling with:

We want to make sure that when a specific page is retrieved for the very first time, it will be generated by the server and served. The second time or more, it will be served by CloudFlare. Currently when you fetch a specific page/category in your browser (initial load), the server generates the document and serves it nicely.

However, when you then navigate in the application, fetch functions may still be executed. We don't want this. We then want the document coming from CloudFlare to be applied.

**What we want to achieve:**

- There are no predefined routes for pages, categories, products. So the routes are dynamic.
- We want an HTML document to be generated when a particular route is consulted for the very first time.
- Subsequently, NUXT has to check every time a route is consulted, whether there is already an HTML document for this. If there is, the cached document must be served. Otherwise it has to render server side. (This must be done both when going directly to a route and when navigating (client)).

**We then have the following additional questions:**

- We want to connect eventual CloudFlare to it. This means that CloudFlare will determine what exactly is served.
- We must take into account that when a certain page/category or product is updated in terms of content, both NUXT and Cloudflare must know that the cache must be deleted and the process starts again once.
- Can someone explain to us in steps what we can do best to achieve this? Or are we overlooking something? Or are there better ways?

**Comment by nhatimme** [comment]
@danielroe: Maybe you can give us and the community a good direction? Thanks mate!

**Comment by warflash** [comment]
I recommend checking out https://discord.com/channels/473401852243869706/1002541842409193563/1105416562481041418, seems like it's pretty much the same question as far as I can tell

**Comment by nhatimme** [comment]
Your link does not work.

**Reply by warflash** [reply]
Just verified, works fine for me. Make sure you are on the nuxt discord otherwise it might not resolve properly
